client/src/components/CardDetail.js

```diff
@@ -0,0 +1,493 @@
+      </Button>
+      <Button
+        variant="contained"
+        className={classes.root}
+        onClick={handleDeadlineClick}
+      >
+        Deadline
+      </Button>
+      <Button
+        variant="contained"
+        className={classes.root}
+        onClick={handleAttachmentClick}
+      >
+        Attachment
+      </Button>
+      <Box mb={2} mt={3}>
+        ACTIONS
+      </Box>
+      <Button variant="contained" className={classes.root}>
+        Move
+      </Button>
+      {/*<Button variant="contained" className={classes.root}>*/}
+      {/*  Copy*/}
+      {/*</Button>*/}
+      {/*<Button variant="contained" className={classes.root}>*/}
+      {/*  Share*/}
+      {/*</Button>*/}
+      <Button variant="contained" className={classes.root} onClick={props.handleDeleteClick}>
+        Delete
+      </Button>
+    </Fragment>
+  );
+};
+
+const ChecklistItems = (props) => {
+  const { checklistItems, setChecklistItems } = props;
+
+  const handleChecklist = (event) => {
+    const newList = [...checklistItems];
+    // const newItem = {
+    //   content:
+    // }
+    newList[event.target.name].active = event.target.checked;
+    setChecklistItems(newList);
+  };
+
+  const handleKeyPress = (event) => {
+    // return;
+    if (event.key === 'Enter'){
+      event.preventDefault();
+      const newList = [...checklistItems];
+      newList.push({content: event.target.value, active: false});
+      setChecklistItems(newList);
+    }
+  }
+
+  return (
+    <FormGroup>
+      {checklistItems.map((item, index) => (
+        <FormControlLabel
+          control={
+            <Checkbox
+              checked={item.active}
+              onChange={handleChecklist}
+              name={index}
+              inputProps={{ 'aria-label': item.content }}
+            />
+          }
+          label={item.content}
+        />
+      ))}
+      <TextField
+        multiline={true}
+        rows={1}
+        type="text"
+        variant={"outlined"}
+        placeholder={"Add an item ..."}
+        size="small"
+        onKeyDown={handleKeyPress}
+      ></TextField>
+    </FormGroup>
+  );
+};
+
+const CardStyle = (theme) => ({
+  root: {
+    height: 700,
+    width: 840,
+  },
+  leftGrid: {
+    [theme.breakpoints.up('sm')] :{
+      width: '79%',
+    },
+    [theme.breakpoints.down('sm')] :{
+      width: '100%',
+    }
+  },
+  rightGrid: {
+    [theme.breakpoints.up('sm')] :{
+      width: '21%',
+    },
+    [theme.breakpoints.down('sm')] :{
+      width: '100%',
+    }
+  },
+  label: {
+    fontSize: 18,
+    fontWeight: "bold",
+    marginBottom: 8,
+  },
+  bottomButton: {
+    height: 34,
+    width: 100,
+    marginRight: theme.spacing(4),
+    marginTop: 20,
+  },
+  datePicker: {
+    textColor: "#759CFC",
+    color: "#759CFC",
+  },
+});
+
+const CardDetail = (props) => {
+  const { handleClose, open } = props;
+  const card = props.card;
+  const {cards, setCards, columns, setColumns} = useDashboard();
```

Comment:
Just as a preference, using the useContext and directly passing the context here makes it more readable when a person non familiar with the codebase reads it